tests/runner.py

```diff
@@ -1368,7 +1393,13 @@ def run_browser(self, html_file, message, expectedResult=None, timeout=None):
           time.sleep(0.1)
         if not received_output:
           BrowserCore.unresponsive_tests += 1
-          print('[unresponsive tests: %d]' % BrowserCore.unresponsive_tests)
+          print('[unresponsive tests: %d, vs %d]' % (BrowserCore.unresponsive_tests, BrowserCore.responsive_tests))
+          BrowserCore.log_browser_harness_status()
+          if BrowserCore.unresponsive_tests == 1 and BrowserCore.responsive_tests == 0:
+            print('[very first test was unresponsive, so maybe the browser just failed to start up; try to restart it]')
+            BrowserCore.restart_browser_harness()
+        else:
+          BrowserCore.responsive_tests += 1
```

Comment:
I believe its normal/fine/recommended to call classmethods via `self` just like normal methods, no?

The factoring of this code makes me think that the browser harness should be a separate class, that should hang of the BrowserCore class.  This would avoid the @classmethod stuff.

e.g.
```
class BrowserHarness():
   def restart(self):
      ..

class BrowserCore():
  # static member, constructed just once at parse time.
  harness = BrowserHarness()

  def test_foo():
     harness.restart()
```

Mostly cosmetic so I won't block this change on that refactor.